measure

```diff
@@ -87,6 +100,113 @@ insights_session = url_session(INSIGHTS_API_URL.format(account_id=NEWRELIC_ACCOU
 insights_session.headers.update({'X-Query-Key': NEWRELIC_INSIGHTS_QUERY_KEY})
 
 
+def aggregate_metric_values_to_scalar(
+    metric_name: str,
+    metrics: Dict[str, Any],
+    result: Dict[str, Any],
+    is_threshold: bool,
+) -> bool:
+    """Recieves information about a single slo condition as well as the fetched metrics and attempts to produce a scalar
+    value computed by producing a list of means from timeseries values returned per instance and producing a scalar from the mean
+    of per instance means
+
+    Args:
+        metric_name (str): Name of the metric to produce a scalar for
+        metrics (Dict[str, Any]): Fetched metrics as returned by get_metrics
+        result (Dict[str, Any]): A reference to a dictionary tracking the result of an individual condition, 
+            used to return the scalar value (keyed metric_scalar/threshold_metric_scalar), information about the metrics payload, 
+            and a message if unable to compute a valid scalar
+        is_threshold (bool): Used to format result info based on whether marshalling a condition metric or threshold_metric
+
+    Returns:
+        bool: True if a scalar was produced, false to indicate the condition relying on the metric must be skipped
+    """
+    preamble = 'Condition threshold' if is_threshold else 'Condition'
+    key_preamble = "threshold_" if is_threshold else ""
+
+    if metric_name not in metrics:
+        result['message'] = f'{preamble} metric not contained in fetched metrics'
+        return False
+
+    metric_values = metrics[metric_name]['values']
+    if not metric_values:
+        result['message'] = f'{preamble} metric contained no instances or values'
```

Comment:
(cosmetic) the 'if' actually checks for 'metric contained empty list of instances'